modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/utils/MapTransformer.scala

```diff
@@ -66,7 +64,7 @@ object MapTransformer {
   type Field = String
 
   // A transformation takes a Key and Value and returns either a failure or anything
-  type TransformFunc = Function2[Key, Value, Either[ValidatorReport, _]]
+  type TransformFunc = Function2[Key, Value, Either[AtomicFieldValidationError, _]]
```

Comment:
```suggestion
  type TransformFunc = Function2[Key, Value, Either[ParseError, _]]
```
(see other comments)